src/components/CustomerQuota/CheckoutUnsuccessful/CheckoutUnsuccessfulCard.tsx

```diff
@@ -0,0 +1,179 @@
+import React, {
+  FunctionComponent,
+  useContext,
+  useEffect,
+  useState,
+} from "react";
+import { View, StyleSheet, ActivityIndicator } from "react-native";
+import { CustomerCard } from "../CustomerCard";
+import { AppText } from "../../Layout/AppText";
+import { color, size } from "../../../common/styles";
+import { sharedStyles } from "../sharedStyles";
+import { FontAwesome } from "@expo/vector-icons";
+import { useTranslate } from "../../../hooks/useTranslate/useTranslate";
+import { Cart } from "../../../hooks/useCart/useCart";
+import { DarkButton } from "../../Layout/Buttons/DarkButton";
+import { AuthContext } from "../../../context/auth";
+import { ShowFullListToggle } from "../ShowFullListToggle";
+import { usePastTransaction } from "../../../hooks/usePastTransaction/usePastTransaction";
+import { Quota } from "../../../types";
+import { TransactionsGroup } from "../TransactionsGroup";
+import { AlertModalContext } from "../../../context/alert";
+import {
+  groupTransactionsByCategory,
+  sortTransactions,
+  getLatestTransactionTime,
+} from "../NoQuota/NoQuotaCard";
+import { CampaignConfigContext } from "../../../context/campaignConfig";
+import { BIG_NUMBER } from "../utils";
+
+const MAX_TRANSACTIONS_TO_DISPLAY = 1;
+
+const styles = StyleSheet.create({
+  checkoutItemsList: {
+    marginTop: size(2),
+  },
+});
+
+interface CheckoutUnsuccessfulCard {
+  ids: string[];
+  cart: Cart;
+  onCancel: () => void;
+  quotaResponse: Quota | undefined;
+}
+
+/**
+ * Shows when the user is not able to return tt-token due to:
+ * - Redeemed token and returned token does not have the same id
+ * - Have not redeemed the token but attempted to return
+ */
+export const CheckoutUnsuccessfulCard: FunctionComponent<CheckoutUnsuccessfulCard> = ({
+  ids,
+  cart,
+  onCancel,
+  quotaResponse,
+}) => {
+  const [isShowFullList, setIsShowFullList] = useState<boolean>(false);
+  const { policies: allProducts } = useContext(CampaignConfigContext);
+  const { sessionToken, endpoint } = useContext(AuthContext);
+  const { pastTransactionsResult, loading, error } = usePastTransaction(
+    ids,
+    sessionToken,
+    endpoint
+  );
+  // Assumes results are already sorted (valid assumption for results from /transactions/history)
+  const sortedTransactions = pastTransactionsResult;
+
+  const { showErrorAlert } = useContext(AlertModalContext);
+  useEffect(() => {
+    if (error) {
+      showErrorAlert(error);
+    }
+  }, [error, showErrorAlert]);
+
+  const latestTransactionTime: Date | undefined =
+    (quotaResponse && getLatestTransactionTime(cart)) ?? undefined;
```

Comment:
Why do we need this given that pastTransactions is already fetched and we can obtain the latest transaction time from there?